frontend/src/app/app-routing.module.ts

```diff
@@ -122,84 +160,114 @@ const routes: Routes = [
       {
         path: 'settings',
+        data: { breadcrumb: 'Settings' },
         children: [
           {
             path: '',
+            data: { breadcrumb: undefined },
             component: SettingsComponent,
           },
           {
             path: 'core',
+            data: { breadcrumb: undefined },
             children: [
               {
                 path: 'users',
+                data: { breadcrumb: 'Users' },
                 component: UserSettingsComponent,
               },
               {
                 path: 'alerts',
+                data: { breadcrumb: 'Alerts' },
                 component: AlertSettingsComponent,
               },
               {
                 path: 'tools',
+                data: { breadcrumb: 'Tools' },
                 children: [
                   {
                     path: '',
+                    data: { breadcrumb: undefined },
                     component: ToolsSettingsComponent,
                   },
                   {
                     path: 'create',
+                    data: { breadcrumb: 'New' },
+                    component: ToolDetailsComponent,
+                  },
+                ],
+              },
+              {
+                path: 'tool/:toolID',
+                data: { breadcrumb: 'Tool', redirect: '/settings/core/tools' },
+                children: [
+                  {
+                    path: '',
+                    data: { breadcrumb: (data: Data) => data.tool?.name },
                     component: ToolDetailsComponent,
                   },
                 ],
               },
-              { path: 'tool/:toolID', component: ToolDetailsComponent },
             ],
           },
           {
             path: 'modelsources',
+            data: { breadcrumb: undefined },
             children: [
               {
                 path: 'git',
+                data: { breadcrumb: 'Git' },
                 children: [
                   {
                     path: '',
+                    data: { breadcrumb: undefined },
                     component: GitSettingsComponent,
                   },
                   {
                     path: 'instances/:id',
+                    data: { breadcrumb: 'Edit' },
                     component: EditGitSettingsComponent,
                   },
                 ],
               },
               {
                 path: 't4c',
+                data: { breadcrumb: 'T4C' },
                 children: [
                   {
                     path: '',
+                    data: { breadcrumb: undefined },
                     component: T4CSettingsComponent,
                   },
                   {
                     path: 'create',
+                    data: { breadcrumb: 'New' },
                     component: EditT4CInstanceComponent,
                   },
                   {
                     path: 'instance/:instance',
+                    data: { breadcrumb: 'Edit' },
```

Comment:
For consistency, I'd show the name instead. We do that for the other router too.